src/libtiled/wangset.cpp

```diff
@@ -53,58 +53,69 @@ void WangSet::addTile(Tile *tile, WangId wangId)
     Q_ASSERT(tile->tileset() == mTileSet);
 
     for (int i = 0; i < 4; ++i) {
-        Q_ASSERT(wangId.getColor(i,true) <= mEdgeColors);
+        Q_ASSERT(wangId.edgeColor(i) <= mEdgeColors);
     }
 
     for (int i = 0; i < 4; ++i) {
-        Q_ASSERT(wangId.getColor(i,false) <= mCornerColors);
+        Q_ASSERT(wangId.cornerColor(i) <= mCornerColors);
     }
 
     mWangIdToTile.insert(wangId, tile);
     mTileIdToWangId.insert(tile->id(), wangId);
 }
 
-Tile *WangSet::getMatchingTile(WangId wangId) const
+Tile *WangSet::findMatchingTile(WangId wangId) const
 {
-    auto potentials = getAllTiles(wangId);
+    auto potentials = findMatchingTiles(wangId);
 
     if (potentials.length() > 0)
         return potentials[qrand() % potentials.length()];
     else
         return NULL;
 }
 
-QList<Tile*> WangSet::getAllTiles(WangId wangId) const
+typedef struct WangWildCard
+{
+    int index, colorCount;
+} WangWildCard;
+
+QList<Tile*> WangSet::findMatchingTiles(WangId wangId) const
 {
     QList<Tile*> list;
 
     //Stores the space of a wild card, followed by how many colors that space can have.
-    QVector<QPoint> wildCards;
+    QVector<WangWildCard> wildCards;
 
     if (mEdgeColors > 0) {
         for (int i = 0; i < 4; ++i) {
-            if (!wangId.getColor(i,true)) {
-                wildCards.append(QPoint(i * 8, mEdgeColors));
+            if (!wangId.edgeColor(i)) {
+                WangWildCard w;
+                w.index = i * 8;
+                w.colorCount = mEdgeColors;
+
+                wildCards.append(w);
             }
         }
     }
 
     if (mCornerColors > 0) {
         for (int i = 0; i < 4; ++i) {
-            if (!wangId.getColor(i,false)) {
-                wildCards.append(QPoint(i * 8 + 4, mCornerColors));
+            if (!wangId.cornerColor(i)) {
+                WangWildCard w;
+                w.index = i * 8 + 4;
+                w.colorCount = mCornerColors;
+
+                wildCards.append(w);
             }
         }
     }
 
     if (wildCards.isEmpty()) {
         list.append(mWangIdToTile.values(wangId));
     } else {
-        QStack<QPoint> stack;
+        QStack<WangWildCard> stack;
 
-        for (int i = 0; i < wildCards.size(); ++i) {
-            stack.push(wildCards[i]);
-        }
+        stack.append(wildCards);
```

Comment:
Right, looking at the Qt 5.4 docs in Qt Creator, the operator+= does work with another QVector, so please just use that instead of append, so we can have it in a single line without raising the Qt dependency.